robot/impl/jobmanager_test.go

```diff
@@ -0,0 +1,1291 @@
+		test.That(tb, logs.FilterMessage("Job failed").Len(),
+			test.ShouldBeGreaterThanOrEqualTo, 1)
+	})
+
+	// get the error log and check the actual message
+	foundErrorLogs := logs.FilterFieldKey("error").All()
+
+	test.That(t, len(foundErrorLogs), test.ShouldBeGreaterThanOrEqualTo, 1)
+	test.That(t, foundErrorLogs[0].ContextMap()["error"], test.ShouldEqual, "DoCommand unimplemented")
+	badResourceJobCfg := config.JobConfig{
+		config.JobConfigData{
+			Name:     "resource not in graph",
+			Schedule: "1s",
+			Resource: "unexpected",
+			Method:   "method",
+		},
+	}
+
+	cfg.Jobs = append(cfg.Jobs, badResourceJobCfg)
+	r.Reconfigure(context.Background(), cfg)
+
+	testutils.WaitForAssertionWithSleep(t, time.Second, 3, func(tb testing.TB) {
+		tb.Helper()
+		test.That(tb, logs.FilterMessageSnippet("Could not get resource").Len(),
+			test.ShouldBeGreaterThanOrEqualTo, 1)
+	})
+
+	// reset logs
+	logs.TakeAll()
+
+	// create functions that return errors
+	model := resource.DefaultModelFamily.WithModel(utils.RandomAlphaString(8))
+	dummyArm := inject.NewArm("arm")
+	dummyArm.GeometriesFunc = func(ctx context.Context) ([]spatialmath.Geometry, error) {
+		return nil, errors.New("test error api function")
+	}
+	dummyArm.DoFunc = func(ctx context.Context, cmd map[string]any) (map[string]any, error) {
+		return nil, errors.New("test error do command")
+	}
+	resource.RegisterComponent(
+		arm.API,
+		model,
+		resource.Registration[arm.Arm, resource.NoNativeConfig]{Constructor: func(
+			ctx context.Context,
+			deps resource.Dependencies,
+			conf resource.Config,
+			logger logging.Logger,
+		) (arm.Arm, error) {
+			return dummyArm, nil
+		}})
+
+	newConf := config.Config{
+		Components: []resource.Config{
+			{
+				Model: model,
+				Name:  "arm",
+				API:   arm.API,
+			},
+		},
+		Jobs: []config.JobConfig{
+			{
+				config.JobConfigData{
+					Name:     "arm job",
+					Schedule: "3s",
+					Resource: "arm",
+					Method:   "GetGeometries",
+				},
+			},
+			{
+				config.JobConfigData{
+					Name:     "arm job do command",
+					Schedule: "3s",
+					Resource: "arm",
+					Method:   "DoCommand",
+					Command: map[string]any{
+						"command": "test",
+					},
+				},
+			},
+		},
+	}
+	r.Reconfigure(context.Background(), &newConf)
+
+	testutils.WaitForAssertionWithSleep(t, time.Second, 5, func(tb testing.TB) {
+		tb.Helper()
+		// check that we have two errors, one from doCommand and one from an rpc call
+		test.That(tb, logs.FilterMessageSnippet("Job failed").Len(),
+			test.ShouldBeGreaterThanOrEqualTo, 2)
+		errorLogs := logs.FilterFieldKey("error").All()
+		errorMessages := []any{}
+		for _, log := range errorLogs {
+			errorMessages = append(errorMessages, log.ContextMap()["error"])
+		}
+		test.That(tb, slices.Contains(errorMessages, "test error do command"), test.ShouldBeTrue)
+		test.That(tb, slices.Contains(errorMessages, "rpc error: code = Unknown desc = test error api function"), test.ShouldBeTrue)
```

Comment:
Out of curiosity, what do the full "job failed" log lines look like for an `InvokeRPC` failure like "test error api function" and for a `DoCommand` failure like "test error do command"?